service/middleware/middleware.go

```diff
@@ -144,6 +149,34 @@ func (s *service) ProcessDuckduckgoSearch(ctx context.Context, m *core.Middlewar
 	return ret, nil
 }
 
+func (s *service) ProcessFetch(ctx context.Context, m *core.Middleware, input string) (*core.MiddlewareProcessResult, error) {
+	// query an URL and return the result
+	val, ok := m.Options["url"]
+	if !ok {
+		return nil, core.ErrBadMiddlewareOptions
+	}
+	url := string(val.(string))
+	req, _ := http.NewRequest("GET", url, nil)
```

Comment:
May need to handle the error, url from the bot configuration, it may be in the wrong format.

By the way, `http.NewRequestWithContext` can replace `http.NewRequest` and `req.WithContext`.